src/server/router/index.js

```diff
@@ -0,0 +1,45 @@
+import qs from 'query-string';
+import React from 'react';
+import { renderToString } from 'react-dom/server';
+import createStore from '../../shared/create-store';
+
+import { match, createMemoryHistory, RouterContext } from 'react-router';
+import { Provider } from 'react-redux';
+import html from '../html';
+
+import getRoutes from '../../shared/routes';
+
+const renderMarkup = (store, routerProps) => {
+  const htmlString = renderToString(
+    <Provider store={store}>
+      <RouterContext {...routerProps}/>
+    </Provider>
+  );
+
+  return html(htmlString, store.getState(), true);
+};
+
+export const requestHandler = (req, res, store, render) => {
+  return (error, redirectLocation, routerProps) => {
+    if (error) {
+      console.error('Routing error:', error);
+      res.status(500).send(error.message);
+    } else if (redirectLocation) {
+      res.redirect(302, redirectLocation.pathname + redirectLocation.search);
```

Comment:
Should this be a permanent `301` redirect opposed to a temporary `302`? Depends on how it's going to be used, from an SEO perspective search engines handle `301` well